weather.py

```diff
@@ -84,13 +86,23 @@ def unique_id(self):
     @property
     def device_info(self):
         """Return the device info."""
-        return {
+        info = {
             "identifiers": {(DOMAIN, self.coordinator.unique_id)},
             "name": NAME,
             "manufacturer": MANUFACTURER,
-            "entry_type": "service",
         }
 
+        # Legacy fallback can be removed when minimum required
+        # HA version is 2021.12.
+        if CURRENT_HA_VERSION >= "2021.12.0b0":
+            from homeassistant.helpers.device_registry import DeviceEntryType
+
+            info["entry_type"] = DeviceEntryType.SERVICE
+        else:
+            info["entry_type"] = "service"
+
+        return info
```

Comment:
This was quite a neat trick, inspired by HACS and https://github.com/hacs/integration/blob/612d5b4e242f9ebea31d65de2c9cc984a3ebbd85/custom_components/hacs/sensor.py#L56.

I have not tested this with HA < 2012.12, but with 2022.2.2 it works fine at least.

Reply:
Looks like HA 2022.3 (released today) did not yet drop the support for this (https://github.com/home-assistant/core/blob/b8861578fffba5c0326643b05855278081a3f939/homeassistant/helpers/device_registry.py#L366). But it would be nice to get this merged and released soon anyway to make sure the integration works with future HA versions 🙂